### PR TITLE
Show `Inf16` (instead of just `Inf`) in `InexactError`

### PR DESCRIPTION
**Pull request by Jay-sanjay**
InexactError shows Inf16 as Inf #51087

I am not sure if it would pass checks further , I will need to review my code again

### CODE REVIEW

**Comment by Seelengrab** [comment]
Thank you for the PR! Please use a descriptive title - it helps tremendously when looking through the list of open PRs to know what a change is about, without having to open it explicitly :)

There also appears to be a doctest failure due to `InexactError` not having a `ex.val` field (see [here](https://buildkite.com/julialang/julia-master/builds/27332#018a571d-9c11-4ae2-8a4f-6cffdd3bdba8/710-744) for the error message, and [here](https://github.com/JuliaLang/julia/blob/8e14322b5aa344639dd86bf9eabb84afe831fcba/base/boot.jl#L343-L347) for the definition of `InexactError`). How this used to work is a mystery to me, to be honest :thinking:

**Comment by Jay-sanjay** [comment]
Yeah sir I was actually in a hurry while making the PR so did not gave a proper name .
 I will go through the doctest failure you provided  and see what else can be done

**Comment by Jay-sanjay** [comment]
@Seelengrab sir can you check please  I was trying to push a commit using the github dev 
image

**Comment by Jay-sanjay** [comment]
I chose the patch-2 branch of my name but it is not getting pushed here , I am unable to get it can you help me out I am a bit new to contribution.

File: base/errorshow.jl [hunk 1]
```diff
@@ -176,7 +176,7 @@ function showerror(io::IO, ex::InexactError)
     T = first(ex.args)
     nameof(T) === ex.func || print(io, T, ", ")
     join(io, ex.args[2:end], ", ")
-    print(io, ")")
+    print(io, repr(ex.val), ')')
```
Comment (KristofferC) [inline base/errorshow.jl:179]: I think this should just use `show(io, ex.val)` and then a `print(io, ')')`.

**Comment by Jay-sanjay** [comment]
@KristofferC  Sir can this me merged now ..?

**Comment by oscardssmith** [comment]
it looks like the tests are failing as a result of this

**Comment by Jay-sanjay** [comment]
bump.

**Comment by Seelengrab** [comment]
The tests still indicate failure here, so as is this can't be merged:

https://buildkite.com/julialang/julia-master/builds/27849#018aa908-ccc5-4b05-a670-875c09213b81/713-747

**Pull request closed without merging**